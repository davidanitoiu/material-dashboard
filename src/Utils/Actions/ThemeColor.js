const SET_THEME_COLOR = 'SET_THEME_COLOR';
const SET_PRIMARY_COLOR = 'SET_PRIMARY_COLOR';
const SET_ACCENT_COLOR = 'SET_ACCENT_COLOR';
const TOGGLE_THEME = 'TOGGLE_THEME';

function setThemeColor(primary, accent) {
    return {
        type: SET_THEME_COLOR,
        primary: primary,
        accent: accent
    };
}

function setPrimaryColor(color) {
    return {
        type: SET_PRIMARY_COLOR,
        color: color,
    };
}

function setAccentColor(color) {
    return {
        type: SET_ACCENT_COLOR,
        color: color,
    };
}

function toggleTheme(){
    return {
        type: TOGGLE_THEME,
    }
}

export {
    SET_THEME_COLOR,
    SET_PRIMARY_COLOR,
    SET_ACCENT_COLOR,
    TOGGLE_THEME,
    setThemeColor,
    setPrimaryColor,
    setAccentColor,
    toggleTheme
}



