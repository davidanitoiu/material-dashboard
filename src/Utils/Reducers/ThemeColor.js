import { createReducer, createAction } from '@reduxjs/toolkit';

export const initialState = {
    themeColor: {
        primary: 'indigo',
        accent: 'yellow',
        type: 'light'
    },
}

const changeMode = mode => mode === 'dark' ? 'light' : 'dark';
export const setPrimaryColor = createAction('themeColor/setPrimary');
export const setAccentColor = createAction('themeColor/setAccent');
export const toggleDark = createAction('themeColor/toggle');

const themeReducer = createReducer(initialState, {
    [setPrimaryColor]: (state, { payload }) => { state.primary = payload },
    [setAccentColor]: (state, { payload }) => { state.accent = payload },
    [toggleDark]: state => { state.type = changeMode(state.type) },
});

export default themeReducer;