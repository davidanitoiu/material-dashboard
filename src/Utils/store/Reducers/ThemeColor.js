import { createReducer, createAction } from '@reduxjs/toolkit';

const initialState = {
    primary: 'indigo',
    accent: 'yellow',
    type: 'light'
}

const changeMode = mode => mode === 'dark' ? 'light' : 'dark';
export const setPrimaryColor = createAction('themeColor/setPrimary');
export const setAccentColor = createAction('themeColor/setAccent');
export const toggleDark = createAction('themeColor/toggle');

export const themeColor = createReducer(initialState, {
    [setPrimaryColor]: (state, { payload }) => { state.primary = payload },
    [setAccentColor]: (state, { payload }) => { state.accent = payload },
    [toggleDark]: state => { state.type = changeMode(state.type) },
});
