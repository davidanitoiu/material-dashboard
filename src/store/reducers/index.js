export * from './themeColor'
export * from './chartContent'
export * from './cardContent'
