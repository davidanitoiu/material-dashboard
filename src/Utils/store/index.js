import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { themeColor } from '@reducers/ThemeColor';
import { cardContent } from '@reducers/CardContent';
import { chartContent } from '@reducers/ChartContent';
import { reduxBatch } from '@manaflair/redux-batch';

const reducer = combineReducers({
    cardContent,
    chartContent,
    themeColor
});

const store = configureStore({
    reducer,
    devTools: process.env.NODE_ENV !== 'production',
    enhancers: [reduxBatch]
})

export default store;