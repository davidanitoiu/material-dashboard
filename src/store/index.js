import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { themeColor,cardContent,chartContent } from './reducers';
import { reduxBatch } from '@manaflair/redux-batch';

const reducer = combineReducers({
    cardContent,
    chartContent,
    themeColor
});

const store = configureStore({
    reducer,
    devTools: process.env.NODE_ENV !== 'production',
    enhancers: [reduxBatch]
})

export default store;