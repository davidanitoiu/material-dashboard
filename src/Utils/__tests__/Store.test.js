import store from '@utils/Store';
import { setPrimaryColor, setAccentColor, toggleDark, changeMode } from '@reducers/ThemeColor';

describe('Redux Store ->', () => {
    it('Store initializes properly #reduxStore', () => {
        expect(store).toMatchSnapshot();
    });
    describe('Theme Color Reducer ->', () => {
        it('When setPrimaryColor dispatch is sent, an action item is generated #reduxStore', () => {
            expect(store.dispatch(setPrimaryColor('green'))).toMatchSnapshot()
        });
        it('When setAccentColor dispatch is sent, an action item is generated #reduxStore', () => {
            expect(store.dispatch(setAccentColor('red'))).toMatchSnapshot()
        });
        it('When toggleDark dispatch is sent, an action item is generated #reduxStore', () => {
            expect(store.dispatch(toggleDark())).toMatchSnapshot();
        });
    })
})