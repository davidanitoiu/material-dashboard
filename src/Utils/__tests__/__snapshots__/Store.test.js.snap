// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux Store -> Store initializes properly #reduxStore 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;

exports[`Redux Store -> Theme Color Reducer -> When setAccentColor dispatch is sent, an action item is generated #reduxStore 1`] = `
Object {
  "payload": "red",
  "type": "themeColor/setAccent",
}
`;

exports[`Redux Store -> Theme Color Reducer -> When setPrimaryColor dispatch is sent, an action item is generated #reduxStore 1`] = `
Object {
  "payload": "green",
  "type": "themeColor/setPrimary",
}
`;

exports[`Redux Store -> Theme Color Reducer -> When toggleDark dispatch is sent, an action item is generated #reduxStore 1`] = `
Object {
  "payload": undefined,
  "type": "themeColor/toggle",
}
`;
